digital_asset_types/src/dao/scopes/asset.rs

```diff
@@ -326,77 +328,107 @@ pub async fn get_related_for_assets(
     if let Some(required) = required_creator {
         assets_map.retain(|_id, asset| asset.creators.iter().any(|c| c.creator == required));
     }
-
     let ids = assets_map.keys().cloned().collect::<Vec<_>>();
-    let authorities = asset_authority::Entity::find()
-        .filter(asset_authority::Column::AssetId.is_in(ids.clone()))
-        .order_by_asc(asset_authority::Column::AssetId)
-        .all(conn)
-        .await?;
+
+    // Fetch asset_data, creators, authorities, and other related data in parallel
+    let (authorities, tokens, attachments, groups) = tokio::join!(
+        // Fetch authorities
+        async {
+            asset_authority::Entity::find()
+                .filter(asset_authority::Column::AssetId.is_in(ids.clone()))
+                .order_by_asc(asset_authority::Column::AssetId)
+                .all(conn)
+                .await
```

Comment:
Just drop the .await since all is a future. You don't need to wrap in sync block.